### PR TITLE
DEX-2013 Prompay via Rapyd APM

### PR DESCRIPTION
**Pull request by d4r1091**
# What this PR does

- Adding `RAPYD_PROMPTPAY` (PromptPay via Rapyd) APM
- Improvement for the `DecodedClientToken` and its parsing for QRCode
- Removed old polling mechanism code
- Minor codebase improvements

Simulator Screen Shot - iPhone 11 - 2022-07-26 at 09 56 11

Simulator Screen Shot - iPhone 11 - 2022-07-26 at 09 56 17

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Extensions & Utilities/UIImageView+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//  PrimerSDK
+//
+//  Copyright © 2022 Primer API ltd. All rights reserved.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+
+extension UIImageView {
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/Extensions & Utilities/UIImageView+Extensions.swift:12]: Same thing, subclass, then extend

File: Sources/PrimerSDK/Classes/Extensions & Utilities/UIImageView+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    
+    func downloaded(from url: URL, contentMode mode: ContentMode = .scaleAspectFit) {
+        contentMode = mode
+        URLSession.shared.dataTask(with: url) { data, response, error in
+            guard
+                let httpURLResponse = response as? HTTPURLResponse, httpURLResponse.statusCode == 200,
+                let mimeType = response?.mimeType, mimeType.hasPrefix("image"),
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/Extensions & Utilities/UIImageView+Extensions.swift:19]: Are you sure they're going to return the mime type? Wouldn't the rest of the checks (converting it to a `UIImage`) suffice?

Comment (d4r1091) [reply Sources/PrimerSDK/Classes/Extensions & Utilities/UIImageView+Extensions.swift]: Addressed ✅

File: Sources/PrimerSDK/Classes/Modules/PollingModule.swift [hunk 3]
```diff
@@ -19,9 +19,11 @@ class PollingModule: Module {
     internal let url: URL
     internal private(set) var isCancelled: Bool = false
     internal var retryInterval: TimeInterval = 3
-    
-    init(url: URL) {
+    private var paymentMethodType: PrimerPaymentMethodType
+
+    init(url: URL, paymentMethodType: PrimerPaymentMethodType) {
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/Modules/PollingModule.swift:24]: As per our discussion :)

**Review by EvansPie — CHANGES_REQUESTED** [review]

**Review by EvansPie — APPROVED** [review]

**Pull request merged**